### ISSUE TITLE
[Feature request]: Warn on "${!var}" without array expansion

### ISSUE THREAD

**Issue by hseg** [koalaman/shellcheck#2858]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
dirs=(foo bar baz)
for i in "${!dirs}"; do
 echo "${dirs[$i]}"
done

```

#### Here's what shellcheck currently says:

Nothing -- passes clean

#### Here's what I wanted or expected to see:

A warning that using indirection on an array is probably not desired,
and that key expansion is probably meant instead.
(the fact that `dirs` is an array seems straightforward to detect)
